commit d976151a51c8e3fd92d49f106430a92a4528ecd6
Author: Heiner Kallweit <hkallweit1@gmail.com>
Date:   Sun Jan 6 20:44:00 2019 +0100

    r8169: don't try to read counters if chip is in a PCI power-save state
    
    [ Upstream commit 10262b0b53666cbc506989b17a3ead1e9c3b43b4 ]
    
    Avoid log spam caused by trying to read counters from the chip whilst
    it is in a PCI power-save state.
    
    Reference: https://bugzilla.kernel.org/show_bug.cgi?id=107421
    
    Fixes: 1ef7286e7f36 ("r8169: Dereference MMIO address immediately before use")
    Signed-off-by: Heiner Kallweit <hkallweit1@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 9fc8a2bc0ff1..b4bca27b460f 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -1730,11 +1730,13 @@ static bool rtl8169_reset_counters(struct rtl8169_private *tp)
 
 static bool rtl8169_update_counters(struct rtl8169_private *tp)
 {
+	u8 val = RTL_R8(tp, ChipCmd);
+
 	/*
 	 * Some chips are unable to dump tally counters when the receiver
-	 * is disabled.
+	 * is disabled. If 0xff chip may be in a PCI power-save state.
 	 */
-	if ((RTL_R8(tp, ChipCmd) & CmdRxEnb) == 0)
+	if (!(val & CmdRxEnb) || val == 0xff)
 		return true;
 
 	return rtl8169_do_counters(tp, CounterDump);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1026_d976151a51c8e3fd92d49f106430a92a4528ecd6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1026_d976151a51c8e3fd92d49f106430a92a4528ecd6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
